core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -16,6 +32,7 @@ export class OptimizedSsrEngine {
   protected currentConcurrency = 0;
   protected renderingCache = new RenderingCache(this.ssrOptions);
   private templateCache = new Map<string, string>();
+  private callbackQueue: Record<string, SsrCallbackFn[] | null> = {};
```

Comment:
What about being more specific that those are _rendering_ callbacks?, e.g. `private renderingCallbacks` or eventually `private renderingCallbackQueues` (but TBH I don't see a need for word queue)

+ a word of JsDoc:
_When the config `reuseCurrentRendering` is true, we want to reuse the html result for all parallel requests of the same URL. Therefore we need to store the callbacks for all those received requests and invoke them with the renderd html, when the initial render outputs the html._